### PR TITLE
Move NATS module with git history

### PR DESCRIPTION
**Pull request by aashikam**
## Purpose
$title

Fixes: https://github.com/ballerina-platform/ballerina-lang/issues/24801

### CODE REVIEW

File: nats-ballerina/Ballerina.toml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+[project]
+org-name = "ballerinax"
+version = "1.0.0"
```
Comment (BuddhiWathsala) [inline nats-ballerina/Ballerina.toml:3]:
```suggestion
version = "@toml.version@"
```

File: nats-native/build.gradle [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    id 'java'
+}
+
+description = 'Ballerina - NATS Java Utils'
+
+dependencies {
+    compile group: 'io.nats', name: 'java-nats-streaming', version: "${natsStreamingVersion}"
+    compile group: 'io.nats', name: 'jnats', version: "${natsVersion}"
+    compile group: 'com.google.protobuf', name: 'protobuf-java', version: "${protobufVersion}"
+    compile group: 'org.ballerinalang', name: 'ballerina-lang', version: "${ballerinaLangVersion}"
```
[the remark this answers is not in the document]
Comment (aashikam) [reply nats-native/build.gradle:28]: Removing it makes `:nats-native:compileJava`  fail. :/

[the remark this answers is not in the document]
Comment (aashikam) [reply nats-native/build.gradle:28]: Used by `AbstractNatsConsumerServiceCompilerPlugin`.

File: compilation-tests/tests/resources/test-src/basic/nats_missing_on_error_parameters.bal [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+}
+service testService on lis {
+     resource function onMessage(nats:StreamingMessage message) {
+        io:println("Received Message");
+    }
+
+    resource function onError() {
+        io:println("Error occurred!");
+    }
+}
```
Comment (daneshk) [inline compilation-tests/tests/resources/test-src/basic/nats_missing_on_error_parameters.bal:35]:
```suggestion
}

```

File: compilation-tests/tests/resources/test-src/basic/nats_missing_on_message_parameter.bal [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+}
+service testService on lis {
+     resource function onMessage() {
+        io:println("Received Message");
+    }
+
+    resource function onError(nats:StreamingMessage message, error errorVal) {
+        io:println("Error occurred!");
+    }
+}
```
Comment (daneshk) [inline compilation-tests/tests/resources/test-src/basic/nats_missing_on_message_parameter.bal:36]:
```suggestion
}

```

File: compilation-tests/tests/resources/test-src/basic/nats_valid_subscriber.bal [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+}
+service testService on lis {
+     resource function onMessage(nats:StreamingMessage message) {
+        io:println("Received Message");
+    }
+
+    resource function onError(nats:StreamingMessage message, error errorVal) {
+        io:println("Error occurred!");
+    }
+}
```
Comment (daneshk) [inline compilation-tests/tests/resources/test-src/basic/nats_valid_subscriber.bal:35]:
```suggestion
}

```

File: gradle/wrapper/gradle-wrapper.properties [hunk 6]
```diff
@@ -0,0 +1,5 @@
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.4-bin.zip
+zipStoreBase=GRADLE_USER_HOME
+zipStorePath=wrapper/dists
```
Comment (daneshk) [inline gradle/wrapper/gradle-wrapper.properties:5]:
```suggestion
zipStorePath=wrapper/dists

```

**Comment by BuddhiWathsala** [comment]
LGTM

**Pull request merged**